src/lib/util/chdcd.cpp

```diff
@@ -1129,6 +1134,437 @@ chd_error chdcd_parse_cue(const char *tocfname, cdrom_toc &outtoc, chdcd_track_i
+			}
+		}
+	}
+
+	fclose(infile);
+
+	return has_rem_singledensity && has_rem_highdensity;
+}
+
+/*-----------------------------------------------------------------
+    chdcd_parse_gdicue - parse a Redump multi-CUE for Dreamcast GDI
+------------------------------------------------------------------*/
+
+/**
+ * @fn  chd_error chdcd_parse_gdicue(const char *tocfname, cdrom_toc &outtoc, chdcd_track_input_info &outinfo)
+ *
+ * @brief   Chdcd parse cue.
+ *
+ * @param   tocfname        The tocfname.
+ * @param [in,out]  outtoc  The outtoc.
+ * @param [in,out]  outinfo The outinfo.
+ *
+ * @return  A chd_error.
+ *
+ * Redump assumes SINGLE-DENSITY starts at 0 LBA and HIGH-DENSITY starts at 45000 LBA. This is *NOT* a generic
+ * multi-cue format its only purpose is Dreamcast GDI dumps. 
+ *
+ * Notes: I'm fairly sure the proper HD start should be 45150 aka 10:02:00. Not 45000 as stated above. However
+ * the old .gdi files in circulation consistently used 45000. This has created an ecosystem of toolchains and
+ * emulators that generate and expect 45000. Using 45150 is tempting but will break compatibility right across
+ * the board. I'm adopting the strategy of behaving identically to .gdi for now.
+ *
+ * Notes: The .gdi format didn't store pregap information. This meant chdman generated .chd files with pregap
+ * and pgtype zeroed out, there was no other choice. Unfortunately now the entire ecosystem of frontends and
+ * emulators which rely on checksums only recognise .chd files with zeroed out pregaps and pgtypes. The brand
+ * new .bin/.cue format does correctly store pregap information and it's even correctly calculated here. But
+ * I can't push that to the .chd without breaking compatibility. So again I'm taking the strategy of behaving
+ * identically to .gdi for now which means zeroing out the pregap and pgtype, as painful as that is.
+ */
+
+chd_error chdcd_parse_gdicue(const char *tocfname, cdrom_toc &outtoc, chdcd_track_input_info &outinfo)
+{
+	FILE *infile;
+	int i, trknum;
+	static char token[512];
+	std::string lastfname;
+	uint32_t wavlen, wavoffs;
+	std::string path = std::string(tocfname);
+	enum gdi_area current_area = SINGLE_DENSITY;
+
+	infile = fopen(tocfname, "rt");
+	path = get_file_path(path);
+	if (infile == (FILE *)nullptr)
+	{
+		return CHDERR_FILE_NOT_FOUND;
+	}
+
+	/* clear structures */
+	memset(&outtoc, 0, sizeof(outtoc));
+	outinfo.reset();
+
+	trknum = -1;
+	wavoffs = wavlen = 0;
+
+	outtoc.flags = CD_FLAG_GDROM;
+
+	while (!feof(infile))
+	{
+		/* get the next line */
+		fgets(linebuffer, 511, infile);
+
+		/* if EOF didn't hit, keep going */
+		if (!feof(infile))
+		{
+			/* single-density area starts LBA = 0 */
+			if (!strncmp(linebuffer, "REM SINGLE-DENSITY AREA", 23)) {
+				current_area = SINGLE_DENSITY;
+				continue;
+			}
+
+			/* high-density area starts LBA = 45000 */
+			if (!strncmp(linebuffer, "REM HIGH-DENSITY AREA", 21)) {
+				current_area = HIGH_DENSITY;
+				continue;
+			}
```

Comment:
Same comment about mixing Allman and K&R styles.

Reply:
Oops, old habits. Resolved in 91af4f306760585ffc48a0bf302144cd81823cd8